app/post_metadata.go

```diff
@@ -53,11 +53,35 @@ func (a *App) PreparePostListForClient(originalList *model.PostList) *model.Post
 	return list
 }
 
+// PostWithOverridenIconURL changes the post icon override URL prop, if it has an emoji icon,
+// so that it points to the URL (relative) of the emoji - static if emoji is default, /api if custom.
+func (a *App) PostWithOverridenIconURL(post *model.Post) *model.Post {
+	prop, ok := post.Props[model.POST_PROPS_OVERRIDE_ICON_EMOJI]
+	if !ok || prop == nil {
+		return post
+	}
+	emojiName := prop.(string)
+	delete(post.Props, model.POST_PROPS_OVERRIDE_ICON_EMOJI)
+
+	if !*a.Config().ServiceSettings.EnablePostIconOverride || emojiName == "" {
+		return post
+	}
+
+	if emojiUrl, err := a.GetEmojiStaticUrl(emojiName); err == nil {
+		post.AddProp(model.POST_PROPS_OVERRIDE_ICON_URL, emojiUrl)
+	} else {
+		mlog.Warn("Failed to retrieve emoji for icon URL: %s.", mlog.String("emojiName", emojiName), mlog.Err(err))
```

Comment:
nit: no need for `%s` here, the parameters are structured. Also, should this say `icon emoji`?
```suggestion
		mlog.Warn("Failed to retrieve URL for override icon emoji", mlog.String("emojiName", emojiName), mlog.Err(err))
```